internal/tiltfile/files.go

```diff
@@ -80,6 +80,19 @@ func (s *tiltfileState) execLocalCmd(t *starlark.Thread, cmd model.Cmd, logOutpu
 		c.Stderr = io.MultiWriter(stderr, logOutput)
 	}
 
+	// if Tilt was invoked with `tilt up --port=XXXXX`, local() calls to use the Tilt API will fail due to trying to
+	// connect to the default port, so explicitly populate the TILT_PORT environment variable if it isn't already
+	hasTiltPort := false
+	for _, e := range c.Env {
+		if strings.HasPrefix(e, "TILT_PORT") {
+			hasTiltPort = true
+			break
+		}
+	}
+	if !hasTiltPort {
+		c.Env = append(c.Env, fmt.Sprintf("TILT_PORT=%d", s.webPort))
```

Comment:
would it be annoying to do TILT_HOST as well?

Reply:
Not really - we've already got `webHost` available on the Tiltfile state (for links display).

I'll add it so that `tilt dump engine` and friends work too.

Reply:
Added in f3d9caa